### PR TITLE
treeSort code added

### PR DESCRIPTION
**Pull request by kobekovich**
Hi Oleksii,

Could you please review the 'tree sort' algorithm?

Thanks

### CODE REVIEW

File: BubbleSort/bubbleSort.js [hunk 1]
```diff
@@ -1,6 +1,22 @@
-let inputArray = [0,-10000,9,7,-200,0,5,1,-999,0];
+let inputArray = [];
+let inputArrayLength = 10000;
+
+function fillInputArray() {
```
Comment (Hopchenko) [inline BubbleSort/bubbleSort.js]: don't repeat yourself, move this function to a separate file and use everywhere you need

File: BubbleSort/bubbleSort.js [hunk 2]
```diff
@@ -1,6 +1,22 @@
-let inputArray = [0,-10000,9,7,-200,0,5,1,-999,0];
+let inputArray = [];
+let inputArrayLength = 10000;
```
Comment (Hopchenko) [inline BubbleSort/bubbleSort.js]: for such kind of values you can use const

File: BubbleSort/bubbleSort.js [hunk 3]
```diff
@@ -1,6 +1,22 @@
...
+
+function fillInputArray() {
+    let multipliers = [-1000,-100,100,1000];
```
Comment (Hopchenko) [inline BubbleSort/bubbleSort.js]: read the difference btw const and let and best practice also

File: LinearSort/linearSort.js [hunk 4]
```diff
@@ -11,15 +25,21 @@ function getMinItem(array, index) {
 }
 
 function sort() {
+    let startTime = new Date();
```
Comment (Hopchenko) [inline LinearSort/linearSort.js]: you can move time measure logic to a separate function to not repeat the code

File: LinearSort/linearSort.js [hunk 5]
```diff
@@ -11,15 +25,21 @@ function getMinItem(array, index) {
...
+
```
Comment (Hopchenko) [inline LinearSort/linearSort.js]: measure function can get measured function as argument

File: TreeSort/treeSort.js [hunk 6]
```diff
@@ -1 +1,83 @@
...
+
+fillInputArray();
+
+let startTime = new Date();
+let tree = new Tree(inputArray[0]);
+insertIntoTree();
+goSort();
+let finishTime = new Date() - startTime;
+
+showResult();
```
Comment (Hopchenko) [inline TreeSort/treeSort.js]: good job!

**Review by Hopchenko — CHANGES_REQUESTED** [review]

File: BubbleSort/bubbleSort.js (hunk 1 above)
Comment (kobekovich) [reply BubbleSort/bubbleSort.js]: fixed

File: BubbleSort/bubbleSort.js (hunk 2 above)
Comment (kobekovich) [reply BubbleSort/bubbleSort.js]: changed to const

File: BubbleSort/bubbleSort.js (hunk 3 above)
Comment (kobekovich) [reply BubbleSort/bubbleSort.js]: ok

File: LinearSort/linearSort.js (hunk 4 above)
Comment (kobekovich) [reply LinearSort/linearSort.js]: done

File: LinearSort/linearSort.js (hunk 5 above)
Comment (kobekovich) [reply LinearSort/linearSort.js]: done

File: TreeSort/treeSort.js (hunk 6 above)
Comment (kobekovich) [reply TreeSort/treeSort.js]: thank you

**Review by Hopchenko — APPROVED** [review]

**Pull request merged**